test/test_macho.rb

```diff
@@ -367,4 +367,44 @@ def test_change_rpath
       delete_if_exists(actual)
     end
   end
+
+  def test_add_rpath
+    pass
+  end
+
+  def test_delete_rpath
+    groups = SINGLE_ARCHES.map do |arch|
+      ["hello.bin", "hello_actual.bin"].map do |fn|
+        fixture(arch, fn)
+      end
+    end
+
+    groups.each do |filename, actual|
+      file = MachO::MachOFile.new(filename)
+
+      refute_empty file.rpaths
+      orig_ncmds = file.ncmds
+      orig_sizeofcmds = file.sizeofcmds
+      orig_npaths = file.rpaths.size
+
+      file.delete_rpath(file.rpaths.first)
+      assert file.ncmds < orig_ncmds
+      assert file.sizeofcmds < orig_sizeofcmds
+      assert file.rpaths.size < orig_npaths
+
+      file.write(actual)
+      # ensure we can actually re-load and parse the modified file
+      modified = MachO::MachOFile.new(actual)
+
+      assert modified
+      assert modified.ncmds < orig_ncmds
+      assert modified.sizeofcmds < orig_sizeofcmds
+      assert modified.rpaths.size == file.rpaths.size
+      assert modified.rpaths.size < orig_npaths
```

Comment:
Maybe add a check that the file size of the original and modified file is still the same? (None of the modifications should alter the file size.)